lib/editing/editor_codemirror.dart

```diff
@@ -240,6 +248,21 @@ class _CodeMirrorDocument extends Document {
 ////      cm.setGutterMarker(an.line - 1, _gutterId,
 ////          _makeMarker(an.type, an.message, an.start, an.end));
     }
+    var path = html.querySelector("#runbutton path");
+    if (containsError) {
+      path.attributes["d"] = "M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M13,13V7H11V13H13M13,17V15H11V17H13Z";
+      path.parent.classes.add("error");
+      path.parent.classes.remove("warning");
+
+   } else if (containsWarning) {
+      path.attributes["d"] = "M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M13,13V7H11V13H13M13,17V15H11V17H13Z";
```

Comment:
It looks like this path / avg icon is the same as the error. Perhaps pull this out into a final local contant in the method?